libs/agno/agno/agent/agent.py

```diff
@@ -1058,13 +1071,22 @@ def run(
         messages: Optional[Sequence[Union[Dict, Message]]] = None,
         stream_intermediate_steps: bool = False,
         retries: Optional[int] = None,
+        knowledge_filters: Optional[Dict[str, Any]] = None,
         **kwargs: Any,
     ) -> Union[RunResponse, Iterator[RunResponse]]:
         """Run the Agent and return the response."""
 
         # Initialize the Agent
         self.initialize_agent()
 
+        self.effective_knowledge_filters = None
+
+        if self.knowledge_filters or knowledge_filters:
+            # initialize metadata (specially required in case when load is commented out)
+            self.knowledge.initialize_valid_filters()  # type: ignore
+
+            self.effective_knowledge_filters = self._get_effective_filters(knowledge_filters)
```

Comment:
As we discussed, this logic can be okay here! But I would try to skip it when not needed. 
Maybe we can add a flag to the knowledge model denoting if the filters are initialized? Like:
```python
if self.knowledge_filters and not self.knowledge.filters_ready:
    [...]
```

Reply:
good idea, lemme try this

Reply:
![image](https://github.com/user-attachments/assets/4689ddc1-6a91-475e-849a-aff2c8aab88c)

@manuhortet how does this look?

![image](https://github.com/user-attachments/assets/9d2ffd0a-599c-4e9d-bce1-ac11c69102b2)